internal/backend/post.go

```diff
@@ -79,5 +85,35 @@ func (d *delphisBackend) NotifySubscribersOfCreatedPost(ctx context.Context, pos
 }
 
 func (d *delphisBackend) GetPostsByDiscussionID(ctx context.Context, discussionID string) ([]*model.Post, error) {
-	return d.db.GetPostsByDiscussionID(ctx, discussionID)
+	iter := d.db.GetPostsByDiscussionIDIter(ctx, discussionID)
+	return d.iterToPosts(ctx, iter)
+	//return d.db.GetPostsByDiscussionID(ctx, discussionID)
+}
+
+// Testing function to keep functionality
+func (d *delphisBackend) iterToPosts(ctx context.Context, iter datastore.PostIter) ([]*model.Post, error) {
+	var posts []*model.Post
+	post := model.Post{}
+
+	for iter.Next(&post) {
+		logrus.Infof("Post: %+v\n", post)
+		tempPost := post
+		if tempPost.QuotedPostID != nil {
+			var err error
```

Comment:
is there a better way to fetch the quoted post? we shouldn't blindly do a join as the vast majority of messages would not have a quote

Reply:
I mean, if the query joins on quotedPostID which also joined on Post, I don't think it's horribly expensive. If the field is `null` it's not going to try and join on that (I don't think) so it will only do it when we want it.

I think that's worth looking into but not high priority if it's a lot of work.

Reply:
discussed using additional queries and will optimize later